docs/src/userguide/nogil.rst

```diff
@@ -0,0 +1,160 @@
+.. _cython_and_gil:
+
+Cython and the GIL
+==================
+
+Python has a global lock (:term:`the GIL <Global Interpreter Lock or GIL>`)
+to ensure that data related to the Python interpreter is not corrupted.
+It is *sometimes* to release this lock in Cython when you are not
+accessing Python data.
+
+There are two occasions when you may want to release the
+GIL:
+
+#. Using :ref:`Cython's parallelism mechanism <parallel>`. The contents of a ``prange`` loop for example are required to be ``nogil``.
+
+#. If you want other (external) Python threads to be able to run at the same time.
+
+    #. if you have a large computationally/IO-intensive block that doesn't need the GIL then it may be "polite" to release it, just to benefit users of your code who want to do multi-threading. However, this is mostly useful rather than necessary.
+
+    #. (very, very occasionally) it's sometimes useful to briefly release the GIL with a short with ``nogil: pass`` block. This is because Cython doesn't release it spontaneously (unlike Python) so if you're waiting on another Python thread to complete a task, this can avoid deadlocks. This sub-point probably doesn't apply to you unless you're compiling GUI code with Cython.
```

Comment:
```suggestion
    #. (very, very occasionally) in long-running Cython code that never calls into the Python interpreter, it can sometimes be useful to briefly release the GIL with a short ``with nogil: pass`` block. This is because Cython doesn't release it spontaneously (unlike the Python interpreter), so if you're waiting on another Python thread to complete a task, this can avoid deadlocks. This sub-point probably doesn't apply to you unless you're compiling GUI code with Cython.
```
Also seems worth wrapping the long lines here to make this more readable and editable.